indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunnerFactory.java

```diff
@@ -46,7 +49,8 @@ public RemoteTaskRunnerFactory(
       final IndexerZkConfig zkPaths,
       final ObjectMapper jsonMapper,
       @Global final HttpClient httpClient,
-      final Supplier<WorkerBehaviorConfig> workerConfigRef
+      final Supplier<WorkerBehaviorConfig> workerConfigRef,
+      ScheduledExecutorFactory factory
```

Comment:
Can this be annotated somehow so that other ScheduledExecutorFactories can be used in the future?